workshops/views.py

```diff
@@ -3154,10 +3147,132 @@ def duplicates(request):
 
     return render(request, 'workshops/duplicates.html', context)
 
+# ------------------------------------------------------------
+# Views for trainees
+
+def _homework2state(homework):
```

Comment:
> This is utility function

Well guess what `util.py` is for :P we used to store some utility functions used by single views over there as well, but we got rid of them (they were too slow & complicated, used in search).